v2/tools/generator/internal/codegen/pipeline/detect_skipping_properties_test.go

```diff
@@ -0,0 +1,168 @@
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// We expect to have four links:
+	// Pkg2020.Person.FullName => Person2020storage.Person.FullName
+	// Pkg2020storage.Person.FullName => Person2021storage.Person.FullName
+	// Pkg2021storage.Person.FullName => Person2022storage.Person.FullName
+	// Pkg2022storage.Person.FullName => EmptyReference
+	g.Expect(detector.links).To(HaveLen(4))
+	AssertLinkExists(g, detector, person2020, test.FullNameProperty.PropertyName())
+	AssertLinkExists(g, detector, person2020s, test.FullNameProperty.PropertyName())
+	AssertLinkExists(g, detector, person2021s, test.FullNameProperty.PropertyName())
+	AssertLinkExists(g, detector, person2022s, test.FullNameProperty.PropertyName())
+}
+
+func TestSkippingPropertyDetector_findBreak_returnsExpectedResults(t *testing.T) {
+
+	// These property references all have different names, for ease of inspection
+	alphaSeen := createPropertyRef("v1", "alpha")
+	betaSeen := createPropertyRef("v2", "beta")
+	gammaSeen := createPropertyRef("v3", "gamma")
+	deltaMissing := createPropertyRef("v4", "delta")
+	epsilonMissing := createPropertyRef("v5", "epsilon")
+	zetaMissing := createPropertyRef("v6", "zeta")
+	thetaSeen := createPropertyRef("v7", "theta")
+	iotaSeen := createPropertyRef("v8", "iota")
+	kappaSeen := createPropertyRef("v9", "kappa")
+	empty := astmodel.EmptyPropertyReference
+
+	types := make(astmodel.Types)
+	cfg := config.NewObjectModelConfiguration()
+	graph, _ := storage.NewConversionGraphBuilder(cfg).Build()
+	detector := NewSkippingPropertyDetector(types, graph)
+
+	detector.addLink(alphaSeen, betaSeen)
+	detector.addLink(betaSeen, gammaSeen)
+	detector.addLink(gammaSeen, deltaMissing)
+	detector.addLink(deltaMissing, epsilonMissing)
+	detector.addLink(epsilonMissing, zetaMissing)
+	detector.addLink(zetaMissing, thetaSeen)
+	detector.addLink(thetaSeen, iotaSeen)
+	detector.addLink(iotaSeen, kappaSeen)
+
+	detector.propertyObserved(alphaSeen)
+	detector.propertyObserved(betaSeen)
+	detector.propertyObserved(gammaSeen)
+	detector.propertyObserved(thetaSeen)
+	detector.propertyObserved(iotaSeen)
+	detector.propertyObserved(kappaSeen)
+
+	cases := []struct {
+		name           string
+		ref            astmodel.PropertyReference
+		expectedBefore astmodel.PropertyReference
+		expectedAfter  astmodel.PropertyReference
+	}{
+		{"At end of chain of observedProperties properties", kappaSeen, kappaSeen, empty},
+		{"When rest of chain all observedProperties", thetaSeen, kappaSeen, empty},
+		{"At end of chain of missing properties", zetaMissing, zetaMissing, thetaSeen},
+		{"In midst of missing properties", deltaMissing, zetaMissing, thetaSeen},
+		{"At end of first run of observedProperties properties", gammaSeen, gammaSeen, deltaMissing},
+		{"In midst of first run of observedProperties properties", betaSeen, gammaSeen, deltaMissing},
+		{"At start of chain of observedProperties properties", alphaSeen, gammaSeen, deltaMissing},
+	}
+
+	t.Parallel()
```

Comment:
minor: I think this usually is put right at the top of the function

Reply:
Moved.